app/src/main/cpp/skyline/kernel/svc.cpp

```diff
@@ -1082,8 +1083,9 @@ namespace skyline::kernel::svc {
                 if (memory->memoryState == memory::states::Heap) {
                     if (memory->guest.data() >= pointer) {
                         if (memory->guest.size() <= size) {
-                            memory->Resize(0);
+                            state.process->memory.FreeMemory(memory->guest);
```

Comment:
You can use freememory at the end of the func with the svc supplied params, no need to use memory objects